code/datums/elements/clickbox.dm

```diff
@@ -0,0 +1,99 @@
+#define CLICKBOX_LAYER FLOAT_LAYER-999
+
+/**
+ * This element adds a near-invisible underlay to the movable target to expand the surface of its icon without
+ * resorting to MOUSE_OPACITY_OPAQUE.
+ * The underlay ignores the parent's color and alpha and can be offset.
+ * The name is a portmanteau of "click" and "hitbox", because technically this isn't an hitbox,
+ * but it helps catch clicks, and I don't want to give it a loooong name like openspace_item_click_handler
+ */
+/datum/element/clickbox
+	element_flags = ELEMENT_DETACH|ELEMENT_BESPOKE
+	id_arg_index = 2
+	var/image/reference_clickbox
+	var/list/mutable_appearances_by_mov = list()
+	/// maximum width/height of the underlay, in case the attached atom is scaled up.
+	var/max_scale = 2
+	/// minimum width/height of the underlay, in case the attached atom is scaled down.
+	var/min_scale = 0.5
+	/// For simple animals that have different sprites when dead.
+	var/dead_state
+
+/datum/element/clickbox/Attach(datum/target, icon_state = "sphere", x_offset = 0, y_offset = 0, max_scale = 2, min_scale = 0.5, dead_state)
+	. = ..()
+	if(!isatom(target))
+		return ELEMENT_INCOMPATIBLE
+	if(!reference_clickbox)
+		reference_clickbox = image(icon = 'icons/misc/clickbox.dmi', icon_state = icon_state, layer = CLICKBOX_LAYER)
+		reference_clickbox.pixel_x = x_offset
+		reference_clickbox.pixel_y = y_offset
+		reference_clickbox.alpha = 1 // Basic
+		reference_clickbox.appearance_flags = RESET_COLOR|RESET_ALPHA
+	if(max_scale)
+		src.max_scale = max_scale
+	if(min_scale)
+		src.min_scale = min_scale
```

Comment:
why check if max_scale and min_scale are defined here? You have defaults set for them where you set your arguments.